src/components/chat/ChatApp.js

```diff
@@ -30,44 +31,52 @@ class ChatApp extends React.Component {
     }
   } // for scrolling
 
-  scrollToBottom = () => {
+  scrollToBottom() {
     this.messagesEndRef.current.scrollIntoView({ behavior: 'smooth' });
-  };
-
-  componentDidMount() {
-    const chatManager = new ChatManager({
-      instanceLocator: INSTANCE_LOCATOR,
-      tokenProvider: new TokenProvider({
-        headers: {
-          Authorization: `Bearer ${this.props.accessToken}`
-        },
-        url: `${BACKEND_URL}/v1/chat/token`
-      }),
-      userId: jwt_decode(this.props.accessToken).sub
-    });
+  }
 
-    chatManager
-      .connect()
-      .then(currentUser => {
-        this.setState({ currentUser });
-        return currentUser.subscribeToRoom({
-          hooks: {
-            onMessage: message => {
-              this.setState({
-                messages: [...this.state.messages, message]
-              });
-            }
+  initialiseChatManager() {
+    try {
+      return new ChatManager({
+        instanceLocator: INSTANCE_LOCATOR,
+        tokenProvider: new TokenProvider({
+          headers: {
+            Authorization: `Bearer ${this.props.accessToken}`
           },
-          messageLimit: 100,
-          roomId: this.props.roomId
-        });
-      })
-      .then(currentRoom => {
-        this.setState({
-          connected: true,
-          currentRoom
-        });
+          url: `${BACKEND_URL}/v1/chat/token`
+        }),
+        userId: jwt_decode(this.props.accessToken).sub
       });
+    } catch (error) {
+      return null;
+    }
+  }
+
+  componentDidMount() {
+    if (this.state.chatManager) {
+      this.state.chatManager
+        .connect()
+        .then(currentUser => {
+          this.setState({ currentUser });
+          return currentUser.subscribeToRoom({
+            hooks: {
+              onMessage: message => {
+                this.setState({
+                  messages: [...this.state.messages, message]
+                });
+              }
+            },
+            messageLimit: 100,
+            roomId: this.props.roomId
+          });
+        })
+        .then(currentRoom => {
+          this.setState({
+            connected: true,
+            currentRoom
+          });
+        });
```

Comment:
If I'm not wrong, you can actually use the [onRejected callback that the Promise provides](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch)

Reply:
I'll take a look later! Thanks

Reply:
I've added a `catch` which does nothing. I was wondering if there's any form of logging available in the frontend? Otherwise there's not much that can be done if chatkit refuses to connect

Reply:
> I've added a `catch` which does nothing. I was wondering if there's any form of logging available in the frontend? Otherwise there's not much that can be done if chatkit refuses to connect

About logging: We have filed an Issue, but no action on this: https://github.com/source-academy/cadet-frontend/issues/169

Reply:
https://github.com/source-academy/cadet-frontend/blob/master/src/utils/notification.ts

 You can use these functions to show errors. The other way is to render an element showing an error

Reply:
Categorised the connection statuses - connecting, connected and failed to connect. Currently rendering a simple error message when connection fails.